src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthSimulateTestsPrecompilesWithRedirection.cs

```diff
@@ -144,17 +144,15 @@ function ecrecover(bytes32 hash, uint8 v, bytes32 r, bytes32 s) public  returns(
         Assert.That(headHash != chain.BlockFinder.Head!.Hash!);
         chain.State.StateRoot = chain.BlockFinder.Head!.StateRoot!;
 
-        TransactionForRpc transactionForRpc = new(new Transaction
+        TransactionForRpc transactionForRpc = TransactionForRpc.FromTransaction(new Transaction
         {
             Data = transactionData,
             To = contractAddress,
             SenderAddress = TestItem.AddressA,
             GasLimit = 3_500_000,
             GasPrice = 20.GWei()
-        })
-        {
-            Nonce = null
-        };
+        });
+        ((LegacyTransactionForRpc)transactionForRpc).Nonce = null;
```

Comment:
It was not on the base type since new Transaction types might not have a `Nonce`. For example, in Optimism we kind of don't have a Nonce (see https://github.com/ethereum-optimism/op-geth/blob/8af19cf20261c0b62f98cc27da3a268f542822ee/core/types/deposit_tx.go#L29-L46)